### PR TITLE
Use a custom makepkg.conf for building and include it in the final image

### PR DESCRIPTION
**Pull request by NeroReflex**
This patch include the makepkg.conf file, used when compiling AUR software in the repository, providing the ability to change how the software is compiled and that same file is included in the resulting image.

For the project this has the benefit of reducing overall build time as in the options check is excluded, therefore resulting executables are not tested.

The rest of the file is equal to the default one I found in the docker container it was used to build chimera.

The main reason I did this is to prevent executing AUR software in the container as that software might be incompatible with the build environment.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -30,6 +30,7 @@ RUN echo -e "keyserver-options auto-key-retrieve" >> /etc/pacman.d/gnupg/gpg.con
 RUN echo -e "#!/bin/bash\nif [[ \"$1\" == \"--version\" ]]; then echo 'fake 244 version'; fi\nmkdir -p /var/cache/pikaur\n" >> /usr/bin/systemd-run && \
     chmod +x /usr/bin/systemd-run
 
+COPY rootfs/etc/makepkg.conf /etc/makepkg.conf
```
Comment (Samsagax) [inline Dockerfile:33]: What is exactly changed from the default? Can it be just a `sed` over the specific line on `makepkg.conf`? My concern is upstream changes can go unnoticed in other areas if we freeze the file in our repo. Might be better to search/replace that specific line.

File: rootfs/etc/makepkg.conf [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+# Makepkg defaults: BUILDENV=(!distcc !color !ccache check !sign)
+#  A negated environment option will do the opposite of the comments below.
+#
+#-- distcc:   Use the Distributed C/C++/ObjC compiler
+#-- color:    Colorize output messages
+#-- ccache:   Use ccache to cache compilation
+#-- check:    Run the check() function if present in the PKGBUILD
+#-- sign:     Generate PGP signature file
+#
+BUILDENV=(!distcc color !ccache check !sign)
```
Comment (Samsagax) [inline rootfs/etc/makepkg.conf:68]: Isn't it supposed to have `!check` here?

**Review by Samsagax — CHANGES_REQUESTED** [review]

File: rootfs/etc/makepkg.conf (hunk 2 above)
Comment (NeroReflex) [reply rootfs/etc/makepkg.conf]: Yes, good spot. I tested this in a fork that targets a zen4 cpu, built with zen3 cpu, so I know disabling check works, but I have forgotten to include that. Updated

File: Dockerfile (hunk 1 above)
Comment (NeroReflex) [reply Dockerfile:36]: I have now included a check that will fail the docker container construction if the upstream makepkg.conf gets changed so it won't go unnoticed.

As for why it is not a simple sed command, well this way the file is in the repo for reference and the process of changing something is straighforward. Moreover, speaking of upstream changes, with sed you will never know if you substituted more than what was intended to just because of upstream change that went unnoticed.

Comment (NeroReflex) [reply Dockerfile:36]: The checksum check additional step refers to a /etc/makepkg.conf file that was calculated from a clean archlinux:base-devel docker container. Once "!check" is restored to "check" the hash will match proving That was the only modification that was done to the one you are currently using.

**Review by Samsagax — CHANGES_REQUESTED** [review]

File: build-image.sh [hunk 3]
```diff
@@ -47,6 +47,9 @@ mkfs.btrfs -f ${BUILD_IMG}
 mount -t btrfs -o loop,nodatacow ${BUILD_IMG} ${MOUNT_PATH}
 btrfs subvolume create ${BUILD_PATH}
 
+# copy the makepkg.conf into chroot
+cp /etc/makepkg.conf rootfs/etc/makepkg.conf
```
Comment (Samsagax) [inline build-image.sh:51]: This is unnecesary. We wouldn't want to force users our configuration.

File: .gitignore [hunk 4]
```diff
@@ -5,4 +5,5 @@
 output/*
 aur-pkgs/*
 !aur-pkgs/build-aur-packages.sh
-!aur-pkgs/build-aur-package.sh
\ No newline at end of file
+!aur-pkgs/build-aur-package.sh
+rootfs/etc/makepkg.conf
```
Comment (Samsagax) [inline .gitignore:9]: Don' t add unrelated changes. Split it up to other PR. Only the sed change should be made here

**Review by Samsagax — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (NeroReflex) [reply .gitignore]: This wasn't meant to end up in this merge request, this means something I did not intended ended up on this branch. I am revoking the merge request
